docker/docker_client.go

```diff
@@ -78,19 +88,21 @@ func newDockerClient(refHostname, certPath string, tlsVerify bool) (*dockerClien
 	}, nil
 }
 
-// makeRequest creates and executes a http.Request with the specified parameters, adding authentication and TLS options for the Docker client.
-// url is NOT an absolute URL, but a path relative to the /v2/ top-level API path.  The host name and schema is taken from the client or autodetected.
-func (c *dockerClient) makeRequest(method, url string, headers map[string][]string, stream io.Reader) (*http.Response, error) {
-	if c.scheme == "" {
-		pr, err := c.ping()
-		if err != nil {
-			return nil, err
+// MakeRequest creates and executes a http.Request with the specified parameters, adding authentication and TLS options for the Docker client.
+// if resolved is fals, eurl is NOT an absolute URL, but a path relative to the /v2/ top-level API path.  The host name and schema is taken from the client or autodetected.
+func (c *dockerClient) MakeRequest(method, url string, headers map[string][]string, stream io.Reader, resolved bool) (*http.Response, error) {
```

Comment:
It seems to me this would be _much_ cleaner to have both `makeRequest` and `makeRequestToResolvedURL` in the interface, or something.  (Really, with no users of the `client` interface, it is hard to tell.)